Sketch/SketchController.m

```diff
@@ -27,7 +34,7 @@ - (void) applicationDidFinishLaunching: (NSNotification *)notif
 	// base located in the bottom left.
 	//[mainItem setFlipped: NO];
 	[mainItem setSize: NSMakeSize(500, 400)];
-	[mainItem setLayout: [ETBezierFreeLayout layout]];
+	[mainItem setLayout: [ETBezierFreeLayout layoutWithObjectGraphContext: [self objectGraphContext]]];
```

Comment:
No, for the first question. The item factory is only for building the UI (the item tree), then in the implementation code of any ETUIObject subclass (e.g. ETLayoutItem, ETLayout, ETBezierPathHandle), the UI object graph should be accessed with self.objectGraphContext.  Similarly if you want to know the object graph of a bezier path handle, you should access it with handle.objectGraphContext and not with -[ETLayoutItemFactory factory].objectGraphContext.

If the UI happens to be cloned or a bezier path handle is copied into another object graph, [ETLayoutItemFactory factory].objectGraphContext won't work anymore, since the copies  will belong to another object graph.

-[ETLayoutItemFactory factory] is just a convenience method to build a non-persistent UI. For persistent UIs, the factory must be created with +factoryWithObjectGraphContext: with  an object graph context that belongs to a COPersistentRoot. Each persistent root represents a distinct document in a CoreObject store.